20_classes_lab/03.py

```diff
@@ -1,13 +1,46 @@
-"""
-Following code assumes a class named Widget that depends on
-other widgets
-Widget ctor takes a name, and function 
-add_dependency(depends)
-marks widget as depends on all widgets in "depends" list
-
-When building a widget, you need to first build all the widgets it
-depends on.
-
-Implement widget so the following code works
-"""
-
+"""
+Classes and objects no 2
+"""
+import sys
+import os
+
+class Widget(object):
+
+    all_objects = []

```

Comment:
And as a bonus, read about weakref and how they would solve some of the problems with `all_objects` class variable:
https://docs.python.org/2/library/weakref.html

(but keep in mind even with weakref - a class variable is still the wrong solution here)